thunder/tests/test_jit_general.py

```diff
@@ -1439,3 +1439,47 @@ def foo(a):
     assert_close(actual, expected)
     assert thunder.cache_misses(jfoo) == 1
     assert thunder.cache_hits(jfoo) == 1
+
+
+@pytest.mark.parametrize(
+    "device",
+    ("cpu", "cuda"),
+)
+def test_cache_symbolic_values_reshape_numel(device):
+    if device == "cuda" and not torch.cuda.is_available():
+        pytest.skip("CUDA not available")
+
+    def foo(a):
+        a = torch.reshape(a, [a.numel()])
+        return a.relu()
+
+    jfoo = thunder.jit(foo, cache="symbolic values")
+
+    a = torch.randn(2, 3, 8, requires_grad=True, device=device)
+
+    actual = jfoo(a)
+    expected = foo(a)
+
+    assert_close(actual, expected)
+
+
+@pytest.mark.parametrize(
+    "device",
+    ("cpu", "cuda"),
+)
+def test_cache_symbolic_values_slice(device):
+    if device == "cuda" and not torch.cuda.is_available():
+        pytest.skip("CUDA not available")
+
+    def foo(a):
+        a = a[..., : a.shape[-1]]
+        return a.relu()
```

Comment:
nit: do we even need the `relu`? would returning the new `a`, or `a.shape` be enough to tickle this code path?

Reply:
The shape unpacking inside `relu` is what triggered the small code change in `producers` (called inside DCE during autograd).

I would like to keep this test as my own guard rail, so I won't accidentally break things in follow up PRs. Should I put a comment to remove them at a later point if you prefer to have leaner tests?

Reply:
nah it's fine, i just didn't realize the necessity. thanks for explaining